builder/azure/arm/step_publish_to_shared_image_gallery.go

```diff
@@ -16,12 +16,26 @@ import (
 
 type StepPublishToSharedImageGallery struct {
 	client  *AzureClient
-	publish func(ctx context.Context, subscriptionID string, sourceID string, sharedImageGallery SharedImageGalleryDestination, miSGImageVersionEndOfLifeDate string, miSGImageVersionExcludeFromLatest bool, miSigReplicaCount int64, location string, diskEncryptionSetId string, tags map[string]string) (string, error)
+	publish func(publishArgs PublishArgs) (string, error)
```

Comment:
Refactored this to make this easier to unit test, and to read as there was a lot of parameters.

Reply:
Nice way to consolidate. I would suggest breaking out the ctx seeing as it is not really a publishing arg and unexported. 
```suggestion
	publish func(ctx context.Context, args PublishArgs) (string, error)
```